File: Algorithm/Sorting/Bubble Sort/bubble_sort.py
from random import randint

def bubble_sort(arr):
    size = len(arr)
    for i in range(size):
        for j in range(i + 1, size):
            if arr[i] > arr[j]:
                arr[i], arr[j] = arr[j], arr[i]

arr = []

for _ in range(20):
    arr.append(randint(0, 100)) # array[20] with random elements from [0, 100]

print("Before: ", arr)
bubble_sort(arr)
print("After : ", arr)

File: Algorithms/Sorting/Bubble Sort/bubbleSort.c
#include<stdio.h>
#include "../array.h"

void bubbleSort(int arr[], int size){
    for (int i = 0; i < size; i++)
    {
        for (int j = i + 1; j < size; j++)
        {
            if(arr[i] > arr[j]){
                swap(&arr[i], &arr[j]); // if elem at the next index is greater, then swap
            }
        }
    }
}

int main()
{
    int arr[] = {48, 45, 27, 29, 10, 18, 20, 37, 47, 13, 29, 28, 39, 32, 35, 14, 32, 11, 42, 38};
    int size = sizeof(arr)/sizeof(int);

    printf("Before sorting: ");
    printArray(arr, size);

    bubbleSort(arr, size);

    printf("After sorting : ");
    printArray(arr, size);
    return 0;
}